src/NexusMods.App/Listeners/NxmRpcListener.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System.Diagnostics;
+using System.Reactive.Linq;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.Extensions.Logging;
+using NexusMods.DataModel.Interprocess;
+using NexusMods.DataModel.Loadouts;
+using NexusMods.DataModel.Loadouts.Markers;
+using NexusMods.Networking.HttpDownloader;
+using NexusMods.Networking.NexusWebApi;
+using NexusMods.Networking.NexusWebApi.DTOs;
+using NexusMods.Networking.NexusWebApi.NMA.Messages;
+using NexusMods.Networking.NexusWebApi.Types;
+using NexusMods.Paths;
+
+namespace NexusMods.App.Listeners;
+
+/// <summary>
+/// Listens to incoming RPC requests to download files.
+/// </summary>
+public class NxmRpcListener : IDisposable
+{
+    private readonly Client _client;
+    private readonly ILogger<NxmRpcListener> _logger;
+    private BufferBlock<NXMUrlMessage> _urlsBlock;
+    private CancellationTokenSource _cancellationSource;
+    private IHttpDownloader _downloader;
+    private readonly TemporaryFileManager _temp;
+
+    // Note: Temporary until we implement the UI part.
+    private LoadoutManager _loadoutManager;
+
+    public NxmRpcListener(IMessageConsumer<NXMUrlMessage> nxmUrlMessages, ILogger<NxmRpcListener> logger, Client client, IHttpDownloader downloader, TemporaryFileManager temp, LoadoutManager loadoutManager)
+    {
+        _logger = logger;
+        _client = client;
+        _downloader = downloader;
+        _temp = temp;
+        _loadoutManager = loadoutManager;
+        _cancellationSource = new CancellationTokenSource();
+        _urlsBlock = new BufferBlock<NXMUrlMessage>();
+        Task.Run(ListenAsync);
```

Comment:
This is a deliberate choice. 

Given task effectively lasts the lifetime of the application; this is not necessary.   
This is effectively a `while(true)` loop; except 99.999% of the time the code will just be yielding to another thread in the threadpool as it awaits the next `nxm://` URI.

I did add a `Dispose` method just in case, that would cancel the task; but for the purposes of completion I don't think that would ever be used at any point.

Reply:
It's still a good approach to wait for the full shutdown, imo. So the despose is something like

```
_cancellationSource.TryCancel()
await _listenTask
```

Also this way we can observe the state of the task if desired, we can tell if it's faulted, still running, etc.

Reply:
Alright, I'll add the code out of principle of following good practices.

Just note the code here is temporary. From next week onward; this code will need to be abstracted; and we're going to most likely have a Task per download which will be submitted to a `DownloadManager` kind of class (which then would be accessible from ViewModels etc.). And all sorts of downloads `e.g.` `nxm://`(s) will be abstracted under a common interface to support features like pausing and cancellation. 

By the end of next sprint most of the code in this class will probably no longer be here.

------------

Misc unrelated note.

We should probably ensure `InstallModsFromArchiveAsync` is cancellation safe; and doesn't leave `DataModel` in an inconsistent state at some point. I might throw up a ticket for this.